src/Database/LSMTree/Internal/Run.hs

```diff
@@ -135,25 +136,17 @@ mkWeakBlobRef Run{runBlobFile} blobspan =
 {-# SPECIALISE close ::
      Run IO h
   -> IO () #-}
--- | Close the files used in the run, but do not remove them from disk.
--- After calling this operation, the run must not be used anymore.
---
--- TODO: Once snapshots are implemented, files should get removed, but for now
--- we want to be able to re-open closed runs from disk.
--- TODO: see openBlobFile DoNotRemoveFileOnClose. This can be dropped at the
--- same once when snapshots are implemented.
+-- | Close the files used in the run and remove them from disk. After calling
+-- this operation, the run must not be used anymore.
 close :: (MonadSTM m, MonadMask m, PrimMonad m) => Run m h -> m ()
 close Run {..} = do
-    -- TODO: removing files should drop them from the page cache, but until we
-    -- have proper snapshotting we are keeping the files around. Because of
-    -- this, we instruct the OS to drop all run-related files from the page
-    -- cache
-    FS.hDropCacheAll runHasBlockIO runKOpsFile
-    FS.hDropCacheAll runHasBlockIO (blobFileHandle runBlobFile)
-
     FS.hClose runHasFS runKOpsFile
-      `finally`
-         BlobFile.removeReference runBlobFile
+    BlobFile.removeReference runBlobFile
+
+    FS.removeFile runHasFS (runKOpsPath runRunFsPaths)
+    FS.removeFile runHasFS (runFilterPath runRunFsPaths)
+    FS.removeFile runHasFS (runIndexPath runRunFsPaths)
+    FS.removeFile runHasFS (runChecksumsPath runRunFsPaths)
```

Comment:
There is no effort any more to handle exceptions here. Do you still want to do that, or at least add a TODO?

Reply:
I've added a TODO